Tasks/GoToolV0/gotool.ts

```diff
@@ -119,20 +116,185 @@ function setGoEnvironmentVariables(goRoot: string) {
+        jsonPath = await toolLib.downloadTool(metadataUrl);
+    } catch (error) {
+        throw new Error(`Failed to download Go releases metadata: ${(error as any).message}`);
+    }
+    const raw = fs.readFileSync(jsonPath, 'utf8');
+
+    let releases: any[];
+    try {
+        releases = JSON.parse(raw);
+    } catch {
+        throw new Error('Failed to parse Go releases metadata from go.dev');
+    }
+
+    let maxPatch = -1;
+    for (const rel of releases) {
+        if (!rel || !rel.version || rel.stable !== true) continue;
+        const v: string = String(rel.version);
+        const m = /^go(\d+)\.(\d+)(?:\.(\d+))?$/i.exec(v);
+        if (!m) continue;
+        const maj = parseInt(m[1], 10);
+        const min = parseInt(m[2], 10);
+        const pat = m[3] ? parseInt(m[3], 10) : 0;
+        if (maj === major && min === minor) {
+            if (pat > maxPatch) maxPatch = pat;
+        }
+    }
+
+    if (maxPatch < 0) {
+        throw new Error(`Could not find a stable patch version for ${major}.${minor} from go.dev`);
+    }
+    return `${major}.${minor}.${maxPatch}`;
+}
+
+async function getMicrosoftLatestPatchFromManifest(majorMinorVersion: string): Promise<string> {
+    const mm = majorMinorVersion.replace(/^go/i, '').replace(/^v/i, '');
+    const manifestUrl = `https://aka.ms/golang/release/latest/go${mm}.assets.json`;
+    tl.debug(`Downloading Microsoft Go manifest from ${manifestUrl}`);
+    let jsonPath: string;
+    try {
+        jsonPath = await toolLib.downloadTool(manifestUrl);
+    } catch (e) {
+        throw new Error(`Failed to download Microsoft Go manifest for ${majorMinorVersion}: ${(e as any).message}`);
+    }
+    const raw = fs.readFileSync(jsonPath, 'utf8');
+
+    let parsed: any;
+    try {
+        parsed = JSON.parse(raw);
+    } catch {
+        parsed = undefined;
+    }
+
+    let full: string | undefined;
+    if (parsed && typeof parsed === 'object') {
+        if (typeof parsed.version === 'string') {
+            full = parsed.version;
+        } else if (Array.isArray(parsed.assets)) {
+            for (const a of parsed.assets) {
+                if (a && typeof a.version === 'string') {
+                    full = a.version;
+                    break;
+                }
+            }
+        }
+    }
+
+    if (!full) {
+        // Fallback regex scan
+        const m = /"version"\s*:\s*"go(\d+\.\d+\.\d+)"/.exec(raw) || /"go(\d+\.\d+\.\d+)"/.exec(raw);
+        if (m) {
+            full = `go${m[1]}`;
+        }
+    }
+
+    if (!full) {
+        throw new Error(`Could not resolve full patch version from Microsoft manifest for ${majorMinorVersion}`);
+    }
+    // Accept versions with or without a leading 'go' prefix and optional prerelease/build metadata
+    // Examples: 1.24.7-1, go1.24.7-rc1, 1.24.7-1+meta, go1.24.7
+    const semverMatch = /^(?:go)?(\d+\.\d+\.\d+(?:-[0-9A-Za-z-.]+)?(?:\+[0-9A-Za-z-.]+)?)$/.exec(full.trim());
```

Comment:
I'm confused--if we have a usable, full version, why is `getMicrosoftLatestPatchFromManifest` being called at all?